kustomization.yaml

```diff
@@ -0,0 +1,80 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+images:
+# Controller
+- name: objectstorage-controller
+  newName: quay.io/containerobjectstorage/objectstorage-controller
+  newTag: latest
+# Sidecar
+- name: sample-driver
+  newName: quay.io/containerobjectstorage/sample-driver
+  newTag: latest
+- name: object-storage-sidecar
+  newName: quay.io/containerobjectstorage/object-storage-sidecar
+  newTag: latest
+
+resources:
+# CRDs
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_bucketaccessclasses.yaml
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_bucketaccesses.yaml
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_bucketaccessrequests.yaml
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_bucketclasses.yaml
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_bucketrequests.yaml
+- https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds/objectstorage.k8s.io_buckets.yaml
+# Controller
+- manifests/ns.yaml
+- manifests/sa.yaml
+- manifests/rbac.yaml
+- manifests/deployment.yaml
+# Sidecar
+- https://raw.githubusercontent.com/container-object-storage-interface/cosi-provisioner-sidecar/master/examples/object-storage-sidecar.yaml
+
+patches:
+# CRDs
+- target:
+    kind: CustomResourceDefinition
+  patch: |-
+    - op: add
+      path: /metadata/annotations
+      value:
+        controller-gen.kubebuilder.io/version: (devel)
+        api-approved.kubernetes.io: https://github.com/kubernetes-sigs/container-object-storage-interface-api/pull/2
+# Controller
+- target:
+    kind: Deployment
+    name: objectstorage-controller
+  patch: |-
+    - op: replace
+      path: /spec/template/spec/containers/0/imagePullPolicy
+      value: IfNotPresent
+# Sidecar
+- target:
+    kind: Deployment
+    name: object-storage-provisioner
+  patch: |-
+    - op: replace
+      path: /spec/template/spec/containers/0/imagePullPolicy
+      value: IfNotPresent
+    - op: replace
+      path: /spec/template/spec/containers/1/imagePullPolicy
+      value: IfNotPresent
+    - op: replace
+      path: /metadata
+      value:
+        name: object-storage-provisioner
+        labels:
+          app: object-storage-provisioner
```

Comment:
This could be coded in the manifest directly, no reason to do it as a patch.

Also, consider adopting https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ instead.